### DISCUSSION TITLE
Pen pressure and documentation

### DISCUSSION THREAD

**Discussion by caseyjp11** [OpenTabletDriver/OpenTabletDriver#1266] [General]
To the devs,

After perusing the bug reports as well as discussion here and additional web searches, it appears that OTD is set up without pen-pressure by default, requiring a plugin (Winink?) to supply it.

I scuffed through the documentation, but there is very little info about it overall, and I assumed that the 'supported' tablets would have pen pressure as the other drivers do. 
I also see links to "OSU", which I'm assuming is an application that uses tablets, but I'm not familiar with that.

I use the tablet for sketching and painting (Krita), teaching myself Blender as well as some Gimp stuff, and having pen pressure is mandatory for what I'm doing.

Before I file a bug report for the lack of pen pressure (out of box, as it were), is this how it is supposed to work? If that is the case, there really needs to be clear documentation if this driver is to get a wider audience.

I have tried the release version (5.3.3-1) as well as the git release as of yesterday with my Huion HS610.
I require all the buttons to be programmable, and in the release version, my tablet is only partially supported with 8 buttons listed.
The GIT version seems to have fixed all of that (thank you for your fast response to my bug report on that), but OTD defaults to making the pen tip "mouse button 1", and when I try other selections the pen movement stops.

I did a little more research and discovered the 'winInk' plugin, but the console reports that it and its parent app are 'out of date' and not loaded for the git master version, and while the plugin loads with 5.3.3-1, the point motion fails as soon as it is loaded.

Anyway, I love the work and effort being put in here, and with a GUI that can actually save settings, it is pretty much what I'm after in a driver set.

I would appreciate a bit more documentation than what is currently available, as this seems to be an outstanding replacement for the hassle of bash scripts with the current non OTD implementation.

System:
Arch linux (updates current)
AMD FX cpu AMD RX500 video
16gig ram
Huion HS610 graphics tablet
tested Opendrivertablet versions, .5.3.3-1   //Git from July14)

addendum:
-----------hid_uclogic is unloaded before starting OTD. I'm concerned that the internal kernel version of digimend might be causing issues, but I'm not a programmer, so I'm not sure where to even begin looking to see if this might be an issue.

Again, thank you for the work on this. 

C

**Comment by InfinityGhost** [comment]
On Linux you should use the Artist Mode to get pen pressure. Windows Ink is Windows only, and its just a workaround for the lack of an API to draw with pen pressure in other apps.

**Reply by caseyjp11** [reply]
Thank you. That did the trick...for the released driver.  :-)

Just an fyi here as I don't normally bug report git 'beta' versions:
 
I reinstalled the git version to play with the interface as it 95% supports the HS610. As soon as I select and apply the artist mode, the pen no longer works. Sometimes the pen will still 'move', but there is no input from pressing and moving the nib (settings appear to be the same, and in absolute mode things do work minus pen pressure.) and other times the tablet quits being recognized.

I checked with the release version and it puts the pen on "mouse button left" (which is correct), and pen pressure '1' as does GIT, but with git...the pen will move but no drawing as it isn't registering for whatever reason. Under the release version, selecting 'artist mode' works. 
This may not be the root issue as once the git release was installed, driver stability went sideways with the pen and mouse conflicting or the tablet just disappearing, requiring systemctl intervention to restart it once 'artist mode' is applied. Journalctl indicated that another driver was accessing the tablet even though hid_uclogic was disabled. Absolute mode works. 

The tablet was disconnected and reconnected to determine if it was the cause, but didn't change  or rectify the problem other than to have the console spit out the aforementioned 'driver in use' for the device.

Anyway, again thanks for the help here, and I look forward to this driver set improving going forward.

Now if we could just get that wheel to work properly. heh.

Last bit: With GIT, bindings is reporting 12 buttons for the HS610, but not reporting the 13th button (on the wheel). xsetwacom reports that this button is available and working (I use xsetwacom to assign a toggle to it using the digimend-git dkms driver as a test.)

**Reply by InfinityGhost** [reply]
Should note that settings are incompatible between v0.5 and git, and it will cause a lot of issues going between them. It's best to reset your settings when switching.

On the note of the 13th button, we plan on adding it as part of wheel support since its more like a wheel button if that makes sense.

**Reply by caseyjp11** [reply]
As an fyi, while testing, I haven't gotten around to 'saving', only applying, as the issues made it moot to try while playing with the drivers. As for switching, I ensure that the opendriver is unloaded and then disabled before replacing it with the git version.
(And unloading and disabling that before using digimend-git again for normal use.) 

tyvm for your support! :-)
C

**Comment by gonX** [comment]
The issue is that we need to simulate a virtual device to expose a virtual tablet - on Windows, this is achieved with the [Windows Ink](https://opentabletdriver.net/Wiki/FAQ/Windows#win-ink) mode, but simulating a virtual device on Windows requires installing a dedicated driver (currently VMulti), which defeats the purpose of the driver being userspace.

So I don't think we'll ever be able to have pen pressure out of the box on Windows unless Windows creates a WindowsInk API similar to `SendInput` which is what's used by the driver on Windows to be completely userspace.

I've opened an issue on our Website tracking that we need to explain this better in the general FAQ at OpenTabletDriver/opentabletdriver.github.io#51
